basic/src/tasks/components/TaskListItem.tsx

```diff
@@ -0,0 +1,48 @@
+import { updateTaskStatus } from "wasp/client/operations";
+import { TagLabel } from "../../tags/components/TagLabel";
+import { TaskWithTags } from "../queries";
+
+interface TaskListItemProps {
+  task: TaskWithTags;
+}
+
+export function TaskListItem({ task }: TaskListItemProps) {
+  async function setTaskDone(
+    event: React.ChangeEvent<HTMLInputElement>,
+  ): Promise<void> {
+    try {
+      await updateTaskStatus({
+        id: task.id,
+        isDone: event.currentTarget.checked,
+      });
+    } catch (err: unknown) {
+      window.alert(`Error while updating task: ${String(err)}`);
+    }
+  }
+
+  return (
+    <li className="group flex w-full flex-wrap items-center justify-between gap-4 rounded-lg px-6 py-3 odd:bg-neutral-50 even:bg-primary-50">
```

Comment:
This is another wild change, but I'd wrap the whole element in a `label`, so when you click on the whole element it acts as the checkbox:
```suggestion
    <li>
      <label
        htmlFor={id}
        className={twJoin(
          "card flex w-full flex-wrap items-center justify-between gap-4 bg-neutral-50 px-6 py-3",
          task.isDone ? "bg-primary-50" : "",
        )}
      >
```

You also don't need the `group` class since you are not styling anything with `group:` styles. Also, I'd use the `bg-primary-50` to indicate the completed tasks (or the other way around) and not make it alternate colors becuase it doesn't convey any extra meaning. The alternate colors pattern is more common in big tables where it helps you to keep track which row you are reading. 


https://github.com/user-attachments/assets/72cc749b-787e-400c-8369-e4809ca8283b

Reply:
Daamn nice improvement.